src/prefect/server/models/flow_runs.py

```diff
@@ -316,6 +316,34 @@ async def read_flow_runs(
     return result.scalars().unique().all()
 
 
+async def cleanup_flow_run_concurrency_slots(
+    session: AsyncSession,
+    deployment_id: UUID,
+    final_state: Optional[State] = None,
+):
```

Comment:
the signature of this utility seems random to me - a `(deployment_id , final_state)` pairing is an ambiguous piece of state; presumably this would be better captured by a `flow_run` object or something?

Reply:
Agreed. The flow run will have its deployment ID and final state.